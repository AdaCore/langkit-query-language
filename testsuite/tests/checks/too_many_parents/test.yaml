driver: 'checker'
rule_name: Too_Many_Parents
input_sources: ['parents.ads']
rule_arguments:
    too_many_parents.n: 2
