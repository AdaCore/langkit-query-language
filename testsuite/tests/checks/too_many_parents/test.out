parents.ads:12:9: rule violation: type has too many parents
12 |    type T_3 is new T_Root and I1 and I2 with null record;   --  FLAG
   |         ^^^

parents.ads:13:9: rule violation: type has too many parents
13 |    type T_4 is new T_Root and I1 and I2 and I3 with null record;   --  FLAG
   |         ^^^

parents.ads:16:19: rule violation: type has too many parents
16 |    protected type PT is new I1 and I2 and I3 with  --  FLAG
   |                   ^^

parents.ads:21:14: rule violation: type has too many parents
21 |    protected P01 is new I1 and I2 and I3 with --  FLAG
   |              ^^^

parents.ads:26:14: rule violation: type has too many parents
26 |    task type TT1 is new I1 and I2 and I3 with  --  FLAG
   |              ^^^

parents.ads:31:9: rule violation: type has too many parents
31 |    task TO1 is new I1 and I2 and I3 with  --  FLAG
   |         ^^^

